common/data_refinery_common/rna_seq.py

```diff
@@ -75,43 +72,40 @@ def should_run_tximport(experiment: Experiment, results, is_tximport_job: bool):
 
 
 def get_quant_results_for_experiment(experiment: Experiment, filter_old_versions=True):
-    """Returns a queryset of salmon quant results from `experiment`."""
+    """Returns a set of salmon quant results from `experiment`."""
     # Subquery to calculate quant results
     # https://docs.djangoproject.com/en/2.2/ref/models/expressions/#subquery-expressions
-
-    # Salmon version gets saved as what salmon outputs, which includes this prefix.
-    current_salmon_version = "salmon " + get_env_variable("SALMON_VERSION", "0.13.1")
+    all_results = ComputationalResult.objects.filter(sample__in=experiment.samples.all())
 
     if filter_old_versions:
-        eligible_results = ComputationalResult.objects.prefetch_related("organism_index").filter(
-            organism_index__salmon_version=current_salmon_version
+        # Salmon version gets saved as what salmon outputs, which includes this prefix.
+        current_salmon_version = "salmon " + get_env_variable("SALMON_VERSION", "0.13.1")
+        organisms = experiment.organisms.all()
+        organism_indices = OrganismIndex.objects.filter(
+            salmon_version=current_salmon_version, organism__in=organisms
         )
-    else:
-        eligible_results = ComputationalResult.objects.all()
+        all_results.filter(organism_index__id__in=organism_indices.values("id"))
 
-    # A result is only eligible to be used if it actually got uploaded.
-    eligible_results = eligible_results.select_related("computedfile").filter(
+    all_results.select_related("computedfile").filter(
         computedfile__s3_bucket__isnull=False, computedfile__s3_key__isnull=False
     )
 
-    # Calculate the computational results sorted that are associated with a given sample (
-    # referenced from the top query)
-    newest_computational_results = eligible_results.filter(
-        samples=OuterRef("id"), processor__name=ProcessorEnum.SALMON_QUANT.value["name"],
-    ).order_by("-created_at")
-
-    # Annotate each sample in the experiment with the id of the most recent computational result
-    computational_results_ids = (
-        experiment.samples.all()
-        .annotate(
-            latest_computational_result_id=Subquery(newest_computational_results.values("id")[:1])
-        )
-        .filter(latest_computational_result_id__isnull=False)
-        .values_list("latest_computational_result_id", flat=True)
-    )
+    def get_sample_id_set(result):
+        return {sample.id for sample in result.samples.all()}
+
+    latest_results = set()
+    for k, group in groupby(sorted(list(all_results), get_sample_id_set), get_sample_id_set):
```

Comment:
`sorted` accepts any iterable, so I don't think you need `list(all_results)`. Also the tests are failing because `sorted` needs `key` to be passed as a kwarg

Reply:
Yeah I figured that out despite the red herring error message. It's weird to me that calling sorted on a set is fine but I guess python just assumes that if you want something sorted it's gotta be in a list.